packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -320,21 +320,40 @@ export default class OptionManager {
    */
   resolvePresets(presets: Array<string | Object>, dirname: string, onResolve?) {
     return presets.map((val) => {
+      let options;
+      if (Array.isArray(val)) {
+        if (val.length > 2) {
+          throw new Error(`Unexpected extra options ${JSON.stringify(val.slice(2))} many ` +
+            "options passed to preset.");
```

Comment:
Ok this is reading weird to me now

`Unexpected extra options asdf, asdf2 many options passed to preset`?